### ISSUE TITLE
[Bug] [Need help to reproduce bug] User profile changed in sidebar on follow/unfollow any user randomly.

### ISSUE THREAD

**Issue by TheAlphamerc** [TheAlphamerc/flutter_twitter_clone#157]
When logged in user follow/unfollow other users from their profile and get back to home page suddenly logged user's profile data on sidebar replaced with other users' profile data. This is happen randomly which is really weird.

**To Reproduce**
Steps to reproduce the behaviour: (Not sure of it)

1. Go to search screen.
2. Go to some users profile page.
3. Click on Follow/unfollow button
4.  Navigate back to search screen.
5. Repeat steps 2,3,4 around 4-5 times.
6. Tap on home page.
7. Open sidebar and check if profile image and profile name changed.

I already followed above steps but didn't able to reproduce bug. Sure thing is the bug exists but didn't know the exact cause.

If anyone facing this issue please let me know. Once I get the steps I'll fix it.
Thanks

**Comment by ublockme** [comment]
But I didn't get this issue yet. I got one issue before, when user share their own profile link and if same user clicks the link, profile must show ```edit profile``` button but for me, it was show follow button for own profile.. but after i update code, now it's ok.

**Comment by ublockme** [comment]
You got this issue when you logging both accounts (from where you follow and which one you follow) in same device right?
I got same issue when i logged both accounts in same device. I think, its because of caches. When we click follow button, Our profile changed to another profile which we logged in already in that device.

**Comment by ublockme** [comment]
i have an idea to solve it. wanna clear cache automatically when user clicks logout button.

**Comment by TheAlphamerc** [comment]
@ublockme it may me possible but not sure. Let me check this.

**Comment by TheAlphamerc** [comment]
I have found the cause of this issue. I have checked for cache initially but all seems good nothing is remain in cache once user logged out. After debugging for a while I found that in `AuthState` class I have implemented a `_onProfileChanged` listener. 

<img width="549" alt="Screenshot 2021-05-30 at 12 21 08 PM" src="https://user-images.githubusercontent.com/37103237/120095093-c1185b80-c0d8-11eb-8e77-c1e2509ee11b.png">

This listener is triggered when logged in user profile data update in firebase database. When this method triggered it return updated user profile data so that we can update user data locally.

Ideally this listener should update whenever a new user logged in app even after existing user logout and login again with other account. And this is what I have forget to implement and in result this issue is arising.

By removing existing listener while logout solved the issue. Now whenever user login in the app after logout it will clear the old listener and new listener will be assigned.

<img width="701" alt="Screenshot 2021-05-30 at 12 28 04 PM" src="https://user-images.githubusercontent.com/37103237/120095283-fbcec380-c0d9-11eb-8b35-eb035748bd78.png">
